src/app/clusters/closure-control-server/closure-control-cluster-logic.cpp

```diff
@@ -392,28 +422,28 @@ CHIP_ERROR ClusterLogic::GetOverallTargetState(DataModel::Nullable<GenericOveral
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR ClusterLogic::GetCurrentErrorList(const AttributeValueEncoder::ListEncodeHelper & encoder)
+CHIP_ERROR ClusterLogic::GetCurrentErrorList(MutableByteSpan & outputSpan)
 {
-    // List can contain at most only 10 Error
-    for (size_t i = 0; i < kCurrentErrorListSize; i++)
-    {
-        ClosureErrorEnum error;
-
-        CHIP_ERROR err = mDelegate.GetCurrentErrorAtIndex(i, error);
-
-        // Convert CHIP_ERROR_PROVIDER_LIST_EXHAUSTED to CHIP_NO_ERROR
-        if (err == CHIP_ERROR_PROVIDER_LIST_EXHAUSTED)
-        {
-            return CHIP_NO_ERROR;
-        }
-
-        // Return for other errors occurred apart from CHIP_ERROR_PROVIDER_LIST_EXHAUSTED
-        ReturnErrorOnFailure(err);
+    assertChipStackLockedByCurrentThread();
+    VerifyOrReturnError(mIsInitialized, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(outputSpan.size() == kCurrentErrorListMaxSize, CHIP_ERROR_BUFFER_TOO_SMALL,
+                        ChipLogError(AppServer, "Output buffer size is not equal to kCurrentErrorListMaxSize"));
+    ByteSpan currentErrorListSpan(reinterpret_cast<const uint8_t *>(mState.mCurrentErrorList),
```

Comment:
This reinterpret_cast is not desirable, and wouldn't be here if the type of the outparam were correct.